tools/fasta_compute_length/fasta_compute_length.xml

```diff
@@ -1,42 +1,74 @@
-<tool id="fasta_compute_length" name="Compute sequence length" version="1.0.1">
+<?xml version="1.0"?>
+<tool id="fasta_compute_length" name="Compute sequence length" version="1.0.2">
     <description></description>
     <command>
+    #if $ref.ref_source == 'dbkey':
+        cp '${ref.index.fields.len_path}' '$output'
+    #else:
         python $__tool_directory__/fasta_compute_length.py
+          #if $ref.ref_source == 'history':
             '$input'
+          #else:
+            '${ref.index.fields.path}'
+          #end if
             '$output'
-            $keep_first
-            $keep_first_word
+            $ref.keep_first
+            $ref.keep_first_word
+    #end if
     </command>
     <inputs>
-        <param name="input" type="data" format="fasta" label="Compute length for these sequences"/>
-        <param name="keep_first" type="integer" value="0" label="How many title characters to keep?" help="'0' = keep the whole thing"/>
-        <param name="keep_first_word" type="boolean" truevalue="id_only" falsevalue="id_and_desc"
-            selected="false" label="Strip fasta description from header?"
-            help="Stripping the description will truncate the fasta header to just the sequence ID. Otherwise the header description will be kept. This step is done before the 'How many characters to keep' option."/>
+        <conditional name="ref">
+            <param name="ref_source" type="select" label="Sequences">
+                <option value="history" selected="True">From History</option>
+                <option value="dbkey">Locally Cached (pre-built length files)</option>
```

Comment:
It is, unfortunately, not that easy. You need to add some boilerplate like tool_data_table_conf files for sample and test data... https://github.com/galaxyproject/tools-iuc/tree/61e1699e65d6fd9f4f73650ed8463b37cd701344/tools/idba_ud

Reply:
And a tool-data folder: https://github.com/galaxyproject/tools-iuc/tree/61e1699e65d6fd9f4f73650ed8463b37cd701344/tools/idba_ud/tool-data

Reply:
Done.